pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -250,7 +252,14 @@ func (r *OAMApplicationReconciler) Reconcile(req reconcile.Request) (reconcile.R
 // ACReconcile contains all the reconcile logic of an AC, it can be used by other controller
 func (r *OAMApplicationReconciler) ACReconcile(ctx context.Context, ac *v1alpha2.ApplicationConfiguration,
 	log logging.Logger) (result reconcile.Result) {
```

Comment:
the key point is we need to refactor the return value here. We should return the reconcileErr in this function, and let the outer logic to handle.